src/shared/settingsConfiguration.ts

```diff
@@ -56,4 +59,38 @@ export class DefaultSettingsConfiguration implements SettingsConfiguration {
             return false
         }
     }
+
+    public readDevSetting<T>(key: AwsDevSetting): string
+    public readDevSetting<T>(key: AwsDevSetting, type: string, silent: boolean): T | undefined
+
+    /**
+     * Gets the value of a developer only setting.
+     *
+     * TODO: show a dialog if this is used, and/or make a UI change (such as
+     * changing the AWS Explorer color) so that it's obvious that Toolkit is in
+     * "Developer mode". Throw an error in CI.
+     */
+    public readDevSetting<T>(key: AwsDevSetting, type: string = 'string', silent: boolean = false): T | undefined {
```

Comment:
Nit: it might be a good idea to adjust the `type` field to the `typeof` acceptable values: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#description